### ISSUE TITLE
Posible memory leak in version 2.3.10-api-22

### ISSUE THREAD

**Issue by hugomenesesp** [dpa99c/cordova-diagnostic-plugin#253]
When signing APK with the new version of Cordova (6.4.0), there's a error because a posible memory leak:

```
:lintVitalRelease<ROUTE_TO_PROJECT>/platforms/android/src/cordova/plugins/Diagnostic.java:167: Error: The WIFI_SERVICE must be looked up on the Application context or memory will leak on devices < Android N. Try changing this.cordova.getActivity() to this.cordova.getActivity().getApplicationContext() [WifiManagerLeak]
                modeName = "unknown";
^
/Users/hugo/Documents/GIT/appmimovistar/platforms/android/src/cordova/plugins/Diagnostic.java:218: Error: The WIFI_SERVICE must be looked up on the Application context or memory will leak on devices < Android N. Try changing this.cordova.getActivity() to this.cordova.getActivity().getApplicationContext() [WifiManagerLeak]
        Intent settingsIntent = new Intent(Settings.ACTION_WIFI_SETTINGS);
                     ^

   Explanation for issues of type "WifiManagerLeak":
   On versions prior to Android N (24), initializing the WifiManager via
   Context#getSystemService can cause a memory leak if the context is not the
   application context. Change context.getSystemService(...) to
   context.getApplicationContext().getSystemService(...).

2 errors, 0 warnings
```

That new version of Cordova uses Gradle 4.1, older versions use 2.14.1

**Comment by hugomenesesp** [comment]
Related Stackoverflow thread: https://stackoverflow.com/a/47319534/1583507

**Comment by dpa99c** [comment]
Your are explicitly using the [v2.3.10-api-22 tag](https://github.com/dpa99c/cordova-diagnostic-plugin/tree/2.3.10-api-22) of the [api-22](https://github.com/dpa99c/cordova-diagnostic-plugin/tree/api-22) branch which is deprecated and (as documented) is no longer maintained.

The [api-22](https://github.com/dpa99c/cordova-diagnostic-plugin/tree/api-22) branch was created at the time that API 23 (Android 6.0 Marshmallow) had just been released and many developers had not yet updated their code to use Android run-time permissions. Android apps should by now be updated to handle run-time permissions on Android and therefore this branch is no longer necessary.

Instead use the [npm releases](https://www.npmjs.com/package/cordova.plugins.diagnostic) from the [master branch](https://github.com/dpa99c/cordova-diagnostic-plugin) which [has been updated](https://github.com/dpa99c/cordova-diagnostic-plugin/commit/8d3a1ccd201b2b982550747fc499af27f3a9b729) to fix the stricter checks under Gradle 4 and so will build fine with `cordova-android@6.4.0`, which targets API 26.

**Comment by hugomenesesp** [comment]
The problem is that my app needs to be compatible from API 19, will it work?

**Comment by dpa99c** [comment]
[cordova-android@6](https://github.com/apache/cordova-android/blob/master/RELEASENOTES.md#600-oct-20-2016):

> Minimum Supported Android Version is Jellybean (API 16/ Android 4.1)

**Comment by dpa99c** [comment]
i.e. both this plugin and `cordova-android@6.4.0` are backwardly-compatible to API 16

**Comment by hugomenesesp** [comment]
The fix should be the same as #251
